tests/composer/composer_store.test.ts

```diff
@@ -1028,6 +1034,28 @@ describe("edition", () => {
     });
   });
 
+  test("Adding a spreading formula at the bottom of the sheet add enough rows for the formula to spread", () => {
+    const sheetId = model.getters.getActiveSheetId();
+    const numberOfRows = model.getters.getNumberRows(sheetId);
+
+    const cellOnLastRow = toXC(0, numberOfRows - 1);
+    editCell(model, cellOnLastRow, "=TRANSPOSE(A1:E1)");
+
+    expect(model.getters.getNumberRows(sheetId)).toBe(numberOfRows + 4 + 50);
+    expect(getCellContent(model, cellOnLastRow)).toBe("0");
+  });
+
+  test("Adding a spreading formula at the right of the sheet add enough cols for the formula to spread", () => {
```

Comment:
You could add test with undos. It would document the known (and accepted) limitation that several history steps are created